lib/iphone-interop-simulator-base.ts

```diff
@@ -0,0 +1,275 @@
+///<reference path="./.d.ts"/>
+"use strict";
+
+import child_process = require("child_process");
+import errors = require("./errors");
+import fs = require("fs");
+import Future = require("fibers/future");
+import options = require("./options");
+import os = require("os");
+import path = require("path");
+import util = require("util");
+import utils = require("./utils");
+
+var $ = require("NodObjC");
+
+export class IPhoneInteropSimulatorBase {
+	constructor(private simulator: IInteropSimulator) { }
+
+	private static FOUNDATION_FRAMEWORK_NAME = "Foundation";
+	private static APPKIT_FRAMEWORK_NAME = "AppKit";
+
+	private static DVT_FOUNDATION_RELATIVE_PATH = "../SharedFrameworks/DVTFoundation.framework";
+	private static DEV_TOOLS_FOUNDATION_RELATIVE_PATH = "../OtherFrameworks/DevToolsFoundation.framework";
+	private static CORE_SIMULATOR_RELATIVE_PATH = "Library/PrivateFrameworks/CoreSimulator.framework";
+	private static SIMULATOR_FRAMEWORK_RELATIVE_PATH_LEGACY = "Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/DVTiPhoneSimulatorRemoteClient.framework";
+	private static SIMULATOR_FRAMEWORK_RELATIVE_PATH = "../SharedFrameworks/DVTiPhoneSimulatorRemoteClient.framework";
+
+	private static DEFAULT_TIMEOUT_IN_SECONDS = 90;
+
+	public run(appPath: string): IFuture<void> {
+		return this.execute(this.launch, { canRunMainLoop: true, appPath: appPath });
+	}
+
+	private launch(appPath: string): void {
+		var sessionDelegate = $.NSObject.extend("DTiPhoneSimulatorSessionDelegate");
+		sessionDelegate.addMethod("session:didEndWithError:", "v@:@@", function(self: any, sel: any, sess: any, error: any) {
+			IPhoneInteropSimulatorBase.logSessionInfo(error, "Session ended without errors.", "Session ended with error ");
+			process.exit(0);
+		});
+		sessionDelegate.addMethod("session:didStart:withError:", "v@:@c@", function(self: any, sel: string, session: any, started: boolean, error:any) {
+			IPhoneInteropSimulatorBase.logSessionInfo(error, "Session started without errors.", "Session started with error ");
+
+			console.log(`${appPath}: ${session("simulatedApplicationPID")}`);
+			if(options.exit) {
+				process.exit(0);
+			}
+		});
+		sessionDelegate.register();
+
+		var appSpec = this.getClassByName("DTiPhoneSimulatorApplicationSpecifier")("specifierWithApplicationPath", $(appPath));
+		var config = this.getClassByName("DTiPhoneSimulatorSessionConfig")("alloc")("init")("autorelease");
+		config("setApplicationToSimulateOnStart",  appSpec);
+		config("setSimulatedApplicationShouldWaitForDebugger", options.waitForDebugger);
+
+		var sdkRoot = options.sdkVersion ? $(this.getSdkRootPathByVersion(options.sdkVersion)) : this.getClassByName("DTiPhoneSimulatorSystemRoot")("defaultRoot");
+		config("setSimulatedSystemRoot", sdkRoot);
+
+		var simulator = this.simulator;
```

Comment:
This code block could be moved into a dedicated function - setupSimulatedDevice, for example